include/wx/msw/window.h

```diff
@@ -734,10 +734,23 @@ class WXDLLIMPEXP_CORE wxWindowMSW : public wxWindowBase
     // this function is called after the window was resized to its new size
     virtual void MSWEndDeferWindowPos()
     {
+        // fix position if limited by Short range
+        if ( PositionIsOutsideShortInterval(m_pendingPosition.x, m_pendingPosition.y) )
+        {
+            MoveOutsideShortInterval(GetHWND(), m_pendingPosition.x, m_pendingPosition.y, m_pendingSize.x, m_pendingSize.y);
+        }
+
         m_pendingPosition = wxDefaultPosition;
         m_pendingSize = wxDefaultSize;
     }
 
+    static inline bool PositionIsOutsideShortInterval(int x, int y)
```

Comment:
AFAICS this shouldn't be inside `#if wxUSE_DEFERRED_SIZING` (it's, of course, a minor concern, as we always use deferred sizing anyhow).